src/components/BirdPopup.tsx

```diff
@@ -0,0 +1,119 @@
+import { BirdPopupProps } from "../types";
+import { useState } from "react";
+import { IoChevronBackOutline, IoChevronForwardOutline } from "react-icons/io5";
+import { useSwipeable } from "react-swipeable";
+
+const BirdPopup: React.FC<BirdPopupProps> = ({ birds, wikiDataMap }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const goToNext = () => {
+    setCurrentIndex((prevIndex) =>
+      prevIndex === birds.length - 1 ? 0 : prevIndex + 1
+    );
+  };
+
+  const goToPrevious = () => {
+    setCurrentIndex((prevIndex) =>
+      prevIndex === 0 ? birds.length - 1 : prevIndex - 1
+    );
+  };
+
+  const swipeHandlers = useSwipeable({
+    onSwipedLeft: goToNext,
+    onSwipedRight: goToPrevious,
+    preventScrollOnSwipe: true,
+    trackMouse: true
+  });
+
+  const currentBird = birds[currentIndex];
+  
+  const wikiData =
+    wikiDataMap.get(currentBird.comName.toUpperCase()) ||
+    // if no bird change eurasian to common and vice versa
+    wikiDataMap.get(
+      currentBird.comName
+        .replace("EURASIAN", "COMMON")
+        .replace("COMMON", "EURASIAN")
+        .toUpperCase()
+    ) ||
+    // if rook go to bird page
+    (currentBird.comName.toUpperCase() === "ROOK"
+      ? wikiDataMap.get("ROOK (BIRD)")
+      : null) ||
+    // if rock pigeon go to rock dove
+    (currentBird.comName.toUpperCase() === "ROCK PIGEON"
+      ? wikiDataMap.get("ROCK DOVE")
+      : null) ||
+      // replace "-" with " "
+      (currentBird.comName.includes("-")
+        ? wikiDataMap.get(currentBird.comName.replace(/-/g, " ").toUpperCase())
+        : null) ||
+    wikiDataMap.get(
+      currentBird.comName.split(" ").slice(1).join(" ").toUpperCase()
+    ) ||
+    Array.from(wikiDataMap.entries()).find(([key]) =>
+      key.includes(
+        currentBird.comName.split(" ").slice(1).join(" ").toUpperCase()
+      )
+    )?.[1];
```

Comment:
The current implementation has redundant computations (e.g., repeated .toUpperCase() and .replace() calls), making it inefficient and harder to maintain. It lacks preprocessing for currentBird.comName, leading to unnecessary recalculations. The nested logic is complex and less readable, increasing the risk of errors when adding or modifying transformations.